taipy/gui/gui.py

```diff
@@ -2907,3 +2907,44 @@ def set_favicon(self, favicon_path: t.Union[str, Path], state: t.Optional[State]
             self._broadcast(
                 "taipy_favicon", url, self._get_client_id() if state else None, message_type=_WsType.FAVICON
             )
+
+    @staticmethod
+    def _add_event_listener(
+        event_name: str,
+        listener: t.Union[
+            t.Callable[[str, t.Dict[str, t.Any]], None], t.Callable[[State, str, t.Dict[str, t.Any]], None]
+        ],
+        with_state: t.Optional[bool] = False,
+    ):
+        try:
+            _Hooks()._add_event_listener(event_name, listener, with_state)
+        except Exception as e:
+            _warn("Hooks: ", e)
+
+    def _fire_event(
+        self, event_name: str, client_id: t.Optional[str] = None, payload: t.Optional[t.Dict[str, t.Any]] = None
+    ):
+        Timer(interval=0.1,
```

Comment:
Plese tell me why the delay...

Reply:
# fire event is executed after 0.1s to let the current state manipulation finish
        # (that would change the credentials for example)
        # in ts, we would use Promise.resolve(() => {__do_fire_event})